### DISCUSSION TITLE
Shopping list - Sync to 3rd party service

### DISCUSSION THREAD

**Discussion by vincentscode** [mealie-recipes/mealie#1838] [Feature Request]
I use [Bring!](https://www.getbring.com/) as my shopping list. I would like to keep it that way but it would be really convenient to sync it to mealie so I could add stuff to my shopping list based on recipes.

For this to be added as a feature it would seem logical to implement it on a more general level, so to support a general sync with several implementations, similar to recipe import.

Bring! in particular provides to ways of integrating with it:
 - [Official Integration](https://www.getbring.com/en/integration-check)
 - [Python API](https://github.com/cschwartz/bring-api)

**Comment by arcreigh** [comment]
Hell it'd be amazing to be able to use a couple different vendors like walmart for example for a quick way to order the relevant items.

**Comment by Niki2k1** [comment]
Would be awesome to have a **Bring!** integration. [Bring Import Docs](https://sites.google.com/getbring.com/bring-import-dev-guide/web-to-app-integration)

**Comment by friko21** [comment]
This would be a really great feature.

**Comment by michael-genson** [comment]
Bring! integration will be possible with https://github.com/mealie-recipes/mealie/pull/3448. Other integrations are supported through the API and Notifier service.

**Reply by vincentscode** [reply]
Hey, that's great to hear! 

This does look like it would require the mealie instance to be exposed to the internet (and the receipe to be public?), in order for Bring! to access and parse the recipe at the mealie URL? Or am I reading this entirely wrong? :smile:

**Reply by michael-genson** [reply]
That's correct, as far as I'm aware Bring doesn't support adding data directly. If you needed to get around this you could write your own service that can take Mealie recipe data and add it to Bring directly (using [something like this](https://github.com/eliasball/python-bring-api)) however Bring has no _official_ support for that.

If that's something you're interested in, the same PR (https://github.com/mealie-recipes/mealie/pull/3448) would enable this using the `POST` integration, though you would have to post to your own custom service and then add it to Bring yourself.

**Reply by vincentscode** [reply]
I see, thanks for the clarification. This does make sense here.

For integrating with a custom service, it would be great if it were possible to receive recipe information in the post body (in order to avoid having to parse the recipe from mealie again with that other service). Is that something that might be possible in the future? Or am I missing some obvious easier way here?

**Reply by michael-genson** [reply]
I could add that to the existing PR, though you can also add query params, e.g. `https://your-endpoint.com?recipeId=${recipeId}`. That's supported in the current version of the PR

**Reply by vincentscode** [reply]
That is true. That would still require the service to "check back" with mealie and possibly have an API token to get structured recipe data back.
For a more general "send-to-bring!"-service it would be nice to not have such a dependency back to mealie.

But I do not think that this is necessarily a feature for this PR (it is a lot of added complexity and most recipe actions will likely not need it for now), just something to keep in mind for the future.

**Reply by michael-genson** [reply]
Yeah that makes sense, should be easy enough to do as well. I can add that to this PR

**Comment by michael-genson** [comment]
https://github.com/mealie-recipes/mealie/pull/3448 has been merged into nightly. For more information on how to integrate with Bring! shopping list or other external sites, [please check out the Mealie docs](https://docs.mealie.io/documentation/getting-started/features/#recipe-actions).
